storm-core/src/jvm/org/apache/storm/pacemaker/Pacemaker.java

```diff
@@ -0,0 +1,374 @@
+                        new MBeanAttributeInfo("total-keys", attributeType, "total-keys", true, false, false)
+        };
+        private PacemakerStats stats;
+
+        public PacemakerDynamicMBean(PacemakerStats stats) {
+            this.stats = stats;
+            this.mBeanInfo = new MBeanInfo("org.apache.storm.pacemaker.PaceMakerDynamicMBean", "Java Pacemaker Dynamic MBean",
+                    PacemakerDynamicMBean.attributeInfos, null, null, null);
+        }
+
+        @Override
+        public MBeanInfo getMBeanInfo() {
+            return mBeanInfo;
+        }
+
+        @Override
+        public AttributeList getAttributes(String[] attributes) {
+            AttributeList list = new AttributeList();
+            if (attributes == null)
+                return list;
+            final int len = attributes.length;
+            try {
+                for (int i = 0; i < len; i++) {
+                    final Attribute a = new Attribute(attributes[i], getAttribute(attributes[i]));
+                    list.add(a);
+
+                }
+            } catch (Exception e) {
+                throw Utils.wrapInRuntime(e);
+            }
+            return list;
+        }
+
+        @Override
+        public Object getAttribute(String attribute) throws AttributeNotFoundException, MBeanException, ReflectionException {
+            if (attribute == null)
+                throw new AttributeNotFoundException("null");
+            if (attribute.equals("send-pulse-count"))
+                return stats.sendPulseCount.get();
+            else if (attribute.equals("total-received-size"))
+                return stats.totalReceivedSize.get();
+            else if (attribute.equals("get-pulse-count"))
+                return stats.getPulseCount.get();
+            else if (attribute.equals("total-sent-size"))
+                return stats.totalSentSize.get();
+            else if (attribute.equals("largest-heartbeat-size"))
+                return stats.largestHeartbeatSize.get();
+            else if (attribute.equals("average-heartbeat-size"))
+                return stats.averageHeartbeatSize.get();
+            else if (attribute.equals("total-keys"))
+                return stats.totalKeys.get();
+            else
+                throw new AttributeNotFoundException("null");
+        }
+
+        @Override
+        public void setAttribute(Attribute attribute) throws AttributeNotFoundException, InvalidAttributeValueException, MBeanException, ReflectionException {
+
+        }
+
+        @Override
+        public AttributeList setAttributes(AttributeList attributes) {
+            return null;
+        }
+
+        @Override
+        public Object invoke(String actionName, Object[] params, String[] signature) throws MBeanException, ReflectionException {
+            return null;
+        }
+    }
+
+    public Pacemaker(Map conf, boolean isRegisterJmx) {
```

Comment:
registering jmx isn't something we want in the constructor. I know stack traces bubble up, but I don't think anything bad actually happens from double registering. If we really want to avoid double-registering, we need a better mechanism than this, but I'm in favor of just letting double-registration occur.